crates/circuit/src/dag_circuit.rs

```diff
@@ -269,7 +269,7 @@ pub struct DAGCircuit {
     var_output_map: _VarIndexMap,
 
     /// Operation kind to count
-    op_names: IndexMap<String, usize, RandomState>,
+    pub op_names: IndexMap<String, usize, RandomState>,
```

Comment:
I think instead of making this public we should be using `count_ops(py, false)`

```suggestion
    op_names: IndexMap<String, usize, RandomState>,
```

Although I can see the value here of having a no copy access method to `op_names` that returns a `&IndexMap` instead of an owned copy like `count_ops` does.